components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java

```diff
@@ -356,6 +339,10 @@ private String resolveUserIdInternal() {
             if (log.isDebugEnabled()) {
                 log.debug("Trying to resolve the user id for the federated user: " + toFullQualifiedUsername());
             }
+            // The federated users from the organization SSO flow will have the user-id set as the username.
+            if (this.isOrganizationUser()) {
+                return this.getUserName();
```

Comment:
Previously, we didn't do this improvement due to the username claim can have user-id with tenant domain appended or user store domain appended or else different identifier as the shared-app subject claim was decided based on the root app's subject claim. But recently we have change it to have the shared-app subject claim as its default behaviour of user-id which won't be overidden by root app's subject claim. https://github.com/wso2-extensions/identity-organization-management/pull/347/files

After that change, we can assure the user-id will be returned correctly, and there is no need to resolve it internally.

Reply:
Only decided set of configurations are allowed to be updated for the shared app. Need to check the subject identifier is allowed to change via API.